ddtrace/contrib/vertica/patch.py

```diff
@@ -0,0 +1,263 @@
+        tags[dbx.USER] = instance.options["user"]
+    if "database" in instance.options:
+        tags[dbx.NAME] = instance.options["database"]
+
+    pin = Pin(
+        service=config.vertica["service_name"],
+        app=APP,
+        app_type=AppTypes.db,
+        tags=tags,
+        _config=config.vertica["patch"]["vertica_python.vertica.cursor.Cursor"],
+    )
+    pin.onto(cursor)
+
+
+# tracing configuration
+config._add(
+    "vertica",
+    {
+        "service_name": "vertica",
+        "app": "vertica",
+        "app_type": "db",
+        "patch": {
+            "vertica_python.vertica.connection.Connection": {
+                "routines": {
+                    "cursor": {"trace_enabled": False, "span_end": cursor_after}
+                }
+            },
+            "vertica_python.vertica.cursor.Cursor": {
+                "routines": {
+                    "execute": {
+                        "operation_name": "vertica.query",
+                        "span_type": "vertica",
+                        "span_start": execute_span_start,
+                        "span_end": execute_span_end,
+                    },
+                    "copy": {
+                        "operation_name": "vertica.copy",
+                        "span_type": "vertica",
+                        "span_start": copy_span_start,
+                    },
+                    "fetchone": {
+                        "operation_name": "vertica.fetchone",
+                        "span_type": "vertica",
+                        "span_end": fetch_span_end,
+                    },
+                    "fetchall": {
+                        "operation_name": "vertica.fetchall",
+                        "span_type": "vertica",
+                        "span_end": fetch_span_end,
+                    },
+                    "nextset": {
+                        "operation_name": "vertica.nextset",
+                        "span_type": "vertica",
+                        "span_end": fetch_span_end,
+                    },
+                }
+            },
+        },
+    },
+)
+
+
+def patch():
+    global _PATCHED
+    if _PATCHED:
+        return
+
+    _install(config.vertica)
+    _PATCHED = True
+
+
+def unpatch():
+    global _PATCHED
+    if _PATCHED:
+        _uninstall(config.vertica)
+        _PATCHED = False
+
+
+def _uninstall(config):
+    for patch_class_path in config["patch"]:
+        patch_mod, _, patch_class = patch_class_path.rpartition(".")
+
+        for patch_routine in config["patch"][patch_class_path]["routines"]:
+            mod = importlib.import_module(patch_mod)
+            cls = getattr(mod, patch_class)
+            unwrap(cls, patch_routine)
+
+
+def _install(config):
+    for patch_class_path in config["patch"]:
+        patch_mod, _, patch_class = patch_class_path.rpartition(".")
+
+        def wrap_init():
+            _patch_item = patch_class_path
+            patch_class_routine = "{}.{}".format(patch_class, "__init__")
```

Comment:
What if we need to patch a module function rather than a class method?

Reply:
@Kyle-Verhoog (idk if that works) note to self: we still need to address this later on.